libr/core/project.c

```diff
@@ -709,42 +712,7 @@ R_API char *r_core_project_notes_file(RCore *core, const char *prj_name) {
 }
 
 R_API bool r_core_project_is_saved(RCore *core) {
-	bool ret;
-	char *saved_dat, *tmp_dat;
-	char *pd = r_str_newf ("%s" R_SYS_DIR "%s",
-		r_config_get (core->config, "dir.projects"),
-		r_config_get (core->config, "prj.name"));
-	if (!pd) {
-		return false;
-	}
-	char *sp = r_str_newf ("%s" R_SYS_DIR "%s", pd, "rc.r2");
-	if (!sp) {
-		free (pd);
-		return false;
-	}
-	char *tsp = r_str_newf ("%s" R_SYS_DIR "tmp", pd);
-	//horrible code follows:
-	free (pd);
-	if (!tsp) {
-		free (sp);
-		return false;
-	}
-	r_core_project_save_script (core, tsp, R_CORE_PRJ_ALL);
-	saved_dat = r_file_slurp (sp, 0);
-	free (sp);
-	if (!saved_dat) {
-		free (tsp);
-		return false;
-	}
-	//Would be better if I knew how to map files in mem in windows
-	tmp_dat = r_file_slurp (tsp, 0);
-	r_file_rm (tsp);
-	free (tsp);
-	if (!tmp_dat) {
-		return false;
-	}
-	ret = !strcmp (tmp_dat, saved_dat);
-	free (tmp_dat);
-	free (saved_dat);
-	return ret;
+	return !core->config->is_dirty 
```

Comment:
i like it! but... is there a way to force dirtyiness to force saving the project ? that would be desirable

Reply:
Yeah in interactive mode we can ask the user to save the project even if no changes was made. And not save if no changes has made